packages/pyright-internal/src/analyzer/sourceFile.ts

```diff
@@ -570,6 +570,15 @@ export class SourceFile {
         try {
             //TODO: this isnt ideal because it re-reads the file for each cell which is unnecessary
             source = getIPythonCells(this.fileSystem, this.getRealUri(), this._console)?.[cellIndex]?.source;
+            // Make sure we don't return any non-python cells
+            if (
+                source &&
+                ['%sql', '%%sql', '%sh', '%pip', '%run', '%fs', '%load', '%matplotlib', '%who', '%env'].includes(
```

Comment:
this only checks the first line of the file. is `%sql` only valid on the first line?  what if there's a newline or something before the line?

is there a way i can run databricks notebooks locally so i can mess around with it myself to learn how they work? it looks like i have to make an account and run it on their site or something?